### ISSUE TITLE
createTypeMap(<interface­>, Class) does not work.

### ISSUE THREAD

**Issue by JSlain** [modelmapper/modelmapper#170]
Creating a mapping with an interface as the source is not getting used.
ModelMapper search for a TypeMap for a (deproxyfied) source.getClass().
If not found, it'll create a default one.
I think it should also search one for all superclasses and/or interfaces before defaulting to a new one.

Here's a test showing the problem:

``` java
    @Test
    public void interfaceAsSource() {
        ModelMapper modelMapper = new ModelMapper();
                                  // Interface
        modelMapper.createTypeMap(Collection.class, HashMap.class).setProvider(
                new Provider<HashMap>() {
                    @Override
                    public HashMap get(ProvisionRequest<HashMap> request) {
                        HashMap<String, String> toReturn = new HashMap<String, String>();
                        toReturn.put("key", "value");
                        return toReturn;
                    }
                });

        Collection<Object> source = new ArrayList<Object>();
        HashMap<?, ?> destination = modelMapper.map(source, HashMap.class);

        assertEquals("value", destination.get("key"));
    }
```

**Comment by chhsiao90** [comment]
Closed by #209

**Comment by chhsiao90** [comment]
The TypeMap inheritance had been release on ModelMapper v1.0.0, please give it a try!
